components/esp_dns/esp_dns_utils.c

```diff
@@ -0,0 +1,211 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <arpa/inet.h>
+#include "esp_random.h"
+#include "esp_dns_utils.h"
+
+/* Function to create a DNS query */
+size_t create_dns_query(uint8_t *buffer, size_t buffer_size, const char *hostname, int addrtype, uint16_t *id_o)
+{
+    /*
+    Sample DNS Query for example.com (Type A)
+    0x00, 0x00,       // Transaction ID
+    0x01, 0x00,       // Flags: Standard query
+    0x00, 0x01,       // Questions: 1
+    0x00, 0x00,       // Answer RRs: 0
+    0x00, 0x00,       // Authority RRs: 0
+    0x00, 0x00,       // Additional RRs: 0
+    0x07, 'e', 'x', 'a', 'm', 'p', 'l', 'e', // QNAME: example.com
+    0x03, 'c', 'o', 'm',
+    0x00,             // End of QNAME
+    0x00, 0x01,       // QTYPE: A (host address)
+    0x00, 0x01        // QCLASS: IN (internet)
+    */
+
+    dns_header_t *header = (dns_header_t *)buffer;
```

Comment:
Is it better to check pointers before using here?

Reply:
This is a private function and the pointer check is performed by the callers. That's why I have not added it here.